Common++/header/MacAddress.h

```diff
@@ -52,61 +57,70 @@ namespace pcpp
 		 *  @param[in] fifthOctet Represent the fifth octet in the address
 		 *  @param[in] sixthOctet Represent the sixth octet in the address
 		 */
-		MacAddress(uint8_t firstOctest, uint8_t secondOctet, uint8_t thirdOctet, uint8_t fourthOctet, uint8_t fifthOctet, uint8_t sixthOctet);
-
-		/**
-		 * A copy constructor for this class
-		 */
-		MacAddress(const MacAddress& other);
+		inline MacAddress(uint8_t firstOctest, uint8_t secondOctet, uint8_t thirdOctet, uint8_t fourthOctet, uint8_t fifthOctet, uint8_t sixthOctet);
 
+#if __cplusplus > 199711L
 		/**
-		 * Overload of the assignment operator
+		 * A constructor that creates an instance out of the initializer list. The length of the list must be equal to 6 (as MAC address is 6-byte long)
+		 * @param[in] addr An initializer list containing the values of type uint8_t representing the MAC address
 		 */
-		MacAddress& operator=(const MacAddress& other);
+		MacAddress(std::initializer_list<uint8_t> octets) : m_IsValid { octets.size() == sizeof m_Address }
+		{
+			if(m_IsValid)
+				std::copy(octets.begin(), octets.end(), std::begin(m_Address));
+		}
+#endif
 
 		/**
 		 * Overload of the comparison operator
 		 * @return true if 2 addresses are equal. False otherwise
 		 */
-		inline bool operator==(const MacAddress& other)
-				{
-					for (int i = 0; i < 6; i++)
-						if (m_Address[i] != other.m_Address[i])
-							return false;
-					return true;
-				}
+		bool operator==(const MacAddress& other) const { return memcmp(m_Address, other.m_Address, sizeof m_Address) == 0; }
 
 		/**
 		 * Overload of the not-equal operator
 		 * @return true if 2 addresses are not equal. False otherwise
 		 */
-		inline bool operator!=(const MacAddress& other) {return !operator==(other);}
+		bool operator!=(const MacAddress& other) const { return !operator==(other); }
+
+#if __cplusplus > 199711L
+		/**
+		 * Overload of the assignment operator
+		 */
+		MacAddress &operator=(std::initializer_list<uint8_t> octets)
+		{
+			m_IsValid = (octets.size() == sizeof m_Address);
+			if(m_IsValid)
+				std::copy_n(std::cbegin(octets), sizeof m_Address, std::begin(m_Address));
+			return *this;
+		}
+#endif
 
 		/**
 		 * Get an indication whether the MAC address is valid. An address can be invalid if it was constructed from illegal input, for example:
 		 * invalid string
 		 * @return True if the address is valid, false otherwise
 		 */
-		bool isValid() { return m_IsValid; }
+		bool isValid() const { return m_IsValid; }
 
 		/**
 		 * Returns a std::string representation of the address
 		 * @return A string representation of the address
 		 */
-		std::string toString();
+		std::string toString() const;
 
 		/**
 		 * Allocates a byte array of length 6 and copies address value into it. Array deallocation is user responsibility
 		 * @param[in] arr A pointer to where array will be allocated
 		 */
-		void copyTo(uint8_t** arr);
+		void copyTo(uint8_t** arr) const;
```

Comment:
If moving most of the method implementations to the header file, why not moving the implementation of this method as well?

Reply:
I'll do it

Reply:
Done